src/ekat/kokkos/ekat_subview_utils.hpp

```diff
@@ -316,6 +315,177 @@ subview_1(const ViewLR<ST******,Props...>& v,
       v.impl_track(),vm);
 }
 
+// ================ Multi-sliced Subviews ======================= //
+// e.g., instead of a single-entry slice like v(:, 42, :), we slice over a range
+// of values, as in v(:, 27:42, :)
+// this means that the subview has the same rank as the "parent" view
+
+// --- Rank1 multi-slice --- //
+template <typename ST, typename... Props>
+KOKKOS_INLINE_FUNCTION
+Unmanaged<ViewLS<ST*, Props...>>
+subview(const ViewLR<ST*, Props...>& v,
+        const Kokkos::pair<int, int> &kp0,
+        const int idim) {
+  assert(v.data() != nullptr);
+  assert(idim == 0);
```

Comment:
I would remove `idim` argument for this case I think, since it is always 0. But ok if you or others disagree.

Reply:
I can't say I have strong opinions either way, other than a potential argument for consistent function signatures. Though I'm not sure that is much of a big deal

Any thought from others?

Reply:
I would remove it. One rule I like when coding is "do not expose parameters/args unless the user is allowed to change/set them". If this parameter cannot be set by the user, then it should not be part of the function signature.

Reply:
So... there's probably something I'm doing wrong, but getting rid of the `idim` argument caused issues with resolving the choice of overloaded function. I'm guessing it's due to handling it within `field_impl.hpp::get_multi_slice_view()` using an if block, i.e.,
```
if (idim == 0 && v_fullsize.rank == 1) {
  return DstView(ekat::subview_m1(v_fullsize,
                              Kokkos::make_pair<int, int>(k_beg, k_end)));
} else {
  return DstView(ekat::subview(
      v_fullsize, Kokkos::make_pair<int, int>(k_beg, k_end), idim));
}
```
In light of that, I'm happy to accept a suggestion for a fix that eliminates the extraneous argument, but in the mean time added a default value for the `idim` argument, and this makes the overloading straightforward again.

Reply:
Yeah, I see now that the way you had it allowed SCREAM side to not have any if-else. 

What is the `ekat::subview_m1()`, is that a typo? If you have an if-else on the SCREAM side, there should be no reason to have the idim input. What was the error you were getting?

If we keep the idim input, I think there needs to still be an assert that the user only uses idim=0, since they could use it thinking it does something else.

Reply:
Ah, it is a type... kind of. That was my attempt to differentiate the two choices by naming them differently. However, my interpretation is that the if/then results in a compile-time ambiguity about which function will be chosen by different sized views, and so doesn't build.

Also, if I'm recalling correctly, when it was just named `ekat::subview(const View<DT*> v, const kk::make_pair<int, int>(a, b))`, the compiler was considering the single-slice 1D function, `ekat::subview(const View<DT*> v, const int idx)`,  as a candidate, which some stranger on the internet seemed to be saying is because the compiler can't decide whether to prioritize choosing a 2-argument overloaded function based on type vs. const-ness 🤷‍♂️

All that being said... I will add that `assert()` back in 🙂